### PR TITLE
Add sample demonstrating chaining ObjectDetector and ObjectTracker skills

### PR DESCRIPTION
**Pull request by dhung-msft**
Add a sample app showcasing chaining ObjectDetector with ObjectTracker skills. The sample periodically runs the detector and initializes trackers based on the detected objects. The ObjectKind filter is applied prior to tracker initialization, i.e. only whitelisted object kinds are tracked.

Tracked objects are arbitrarily assigned numbers on initialization. The number of an object may change at tracker re-initialization (when the detector runs); there is no object correlation or tracker state preserved through detector evaluation/tracker re-initialization

### CODE REVIEW

File: samples/ChainedSkills/cs/ChainedSkillsSample_UWP/Package.appxmanifest [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Package xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10" xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest" xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10" IgnorableNamespaces="uap mp">
+  <Identity Name="1eafd30a-b4c8-4cfa-b0f5-15790b839905" Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" Version="1.0.0.0" />
```
Comment (LPBourret) [inline samples/ChainedSkills/cs/ChainedSkillsSample_UWP/Package.appxmanifest]: >1eafd30a-b4c8-4cfa-b0f5-15790b839905 [](start = 18, length = 36)

already in use by the object detector sample

File: samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+            // Ready to begin, enable buttons
+            NotifyUser("Skill initialized. Select a media source from the top to begin.");
+            await UpdateMediaSourceButtonsAsync(true);
+        }
+
+        /// <summary>
+        /// Populate UI with skill information and options
+        /// </summary>
+        /// <returns></returns>
+        private async Task UpdateSkillUIAsync()
```
Comment (LPBourret) [inline samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs]: > private async Task UpdateSkillUIAsync() [](start = 7, length = 40)

segment per skill used here or don't show..

File: samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+                    // Display available execution devices
+                    UISkillExecutionDevices.ItemsSource = m_availableExecutionDevices.Select((device) => $"{device.ExecutionDeviceKind} | {device.Name}");
+                    
+                    // Set SelectedIndex to index of currently selected device
+                    for (int i = 0; i < m_availableExecutionDevices.Count; i++)
+                    {
+                        if (m_availableExecutionDevices[i].ExecutionDeviceKind == m_detectorBinding.Device.ExecutionDeviceKind
+                            && m_availableExecutionDevices[i].Name == m_detectorBinding.Device.Name)
+                        {
+                            UISkillExecutionDevices.SelectedIndex = i;
```
Comment (LPBourret) [inline samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs]: >UISkillExecutionDevices [](start = 28, length = 23)

do a inner join between the 2 skills..

File: samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,617 @@
...
+                    //// Retrieve and filter results if requested
+                    //IReadOnlyList<ObjectDetectorResult> objectDetections = m_detectorBinding.DetectedObjects;
+                    //if (m_objectKinds?.Count > 0)
+                    //{
+                    //    objectDetections = objectDetections.Where(det => m_objectKinds.Contains(det.Kind)).ToList();
+                    //}
+
+                    //// Update displayed results
+                    //// FIXME
+                    //m_renderer.Render(objectDetections);
```
Comment (LPBourret) [inline samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs]: remove

File: samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,617 @@
...
+                // Create new frame source and register a callback if the source fails along the way
+                m_frameSource = await FrameSourceFactory.CreateFrameSourceAsync(
+                    source, 
+                    (sender, message) => 
+                    {
+                        NotifyUser(message);
+                    },
+                    inputImageDescriptor);
+
+                // TODO: Workaround for a bug in ObjectDetectorBinding when binding consecutively VideoFrames with Direct3DSurface and SoftwareBitmap
```
Comment (LPBourret) [inline samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs]: >Workaround for a bug in ObjectDetectorBinding when binding consecutively VideoFrames [](start = 25, length = 84)

is this still the case?

File: samples/ChainedSkills/cs/ChainedSkillsSample_UWP/Package.appxmanifest [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  </Properties>
+  <Dependencies>
+    <TargetDeviceFamily Name="Windows.Universal" MinVersion="10.0.0.0" MaxVersionTested="10.0.0.0" />
+  </Dependencies>
+  <Resources>
+    <Resource Language="x-generate" />
+  </Resources>
+  <Applications>
+    <Application Id="App" Executable="$targetnametoken$.exe" EntryPoint="ChainedSkillsSample.App">
+      <uap:VisualElements DisplayName="ChainedSkillsSample" Square150x150Logo="Assets\Square150x150Logo.png" Square44x44Logo="Assets\Square44x44Logo.png" Description="ChainedSkillsSample" BackgroundColor="transparent">
```
Comment (LPBourret) [inline samples/ChainedSkills/cs/ChainedSkillsSample_UWP/Package.appxmanifest]: >ChainedSkillsSample [](start = 39, length = 19)

maybe something more evocative would be better (i.e. DetectAndTrackObjectsSample, ObjectDetectAndTrackSample, etc.)

File: samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs (hunk 5 above)
Comment (dhung-msft) [reply samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs]: I couldn't find an internal bug matching this (must've forgotten to file one) and it doesn't appear to repro on 19H1, so I think no?

File: samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,664 @@
...
+                    {
+                        if (m_detectorExecutionDevices[i].ExecutionDeviceKind == m_detectorBinding.Device.ExecutionDeviceKind
+                            && m_detectorExecutionDevices[i].Name == m_detectorBinding.Device.Name)
+                        {
+                            UIDetectorExecutionDevices.SelectedIndex = i;
+                            break;
+                        }
+                    }
+
+                    // Display available detector execution devices
```
Comment (LPBourret) [inline samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs]: >detector [](start = 41, length = 8)

tracker

File: samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,664 @@
...
+            {
+                m_detectorSkill = await m_detectorDescriptor.CreateSkillAsync() as ObjectDetectorSkill;
+            }
+            m_detectorBinding = await m_detectorSkill.CreateSkillBindingAsync() as ObjectDetectorBinding;
+
+            m_inputImageFeatureDescriptor = m_detectorBinding["InputImage"].Descriptor as SkillFeatureImageDescriptor;
+        }
+
+        /// <summary>
+        /// Initialize the ObjectDetector skill
```
Comment (LPBourret) [inline samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs]: >ObjectDetector [](start = 27, length = 14)

ObjectTracker

File: samples/CombinedSkillsSamples/cs/DetectAndTrackObjectsSample_UWP/MainPage.xaml [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+                               BorderThickness="1"
+                               Margin="0,5"
+                               IsExpanded="False"
+                               ExpandDirection="Down"
+                               Expanded="UIExpander_Expanded"
+                               Collapsed="UIExpander_Expanded">
+                            <ListView Name="UIObjectKindFilters" SelectionMode="Multiple" ScrollViewer.VerticalScrollBarVisibility="Visible" SelectionChanged="UIObjectKindFilters_SelectionChanged" />
+                        </controls:Expander>
+                        <StackPanel Orientation="Horizontal">
+                            <TextBox Width="200" Margin="0,0,5,0" Header="Detector Evaluation Interval" TextChanging="ValidateNumericalTextBoxEntry" Name="UIDetectorIntervalBox" />
```
Comment (LPBourret) [inline samples/CombinedSkillsSamples/cs/DetectAndTrackObjectsSample_UWP/MainPage.xaml]: >ValidateNumericalTextBoxEntry [](start = 118, length = 29)

suggestion: use the Community toolkit extensions for specifying regex validation on textbox entry: [https://docs.microsoft.com/en-us/windows/communitytoolkit/extensions/textboxregex](https://docs.microsoft.com/en-us/windows/communitytoolkit/extensions/textboxregex)

File: samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs (hunk 7 above)
Comment (LPBourret) [reply samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs]: nit: still there :p

---
In reply to: [335700595](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/60#discussion_r335700595) [](ancestors = 335700595)

Comment (dhung-msft) [reply samples/ChainedSkills/cs/ChainedSkillsSample_UWP/MainPage.xaml.cs]: Shouldn't be, are you checking latest?

**Pull request merged**